src/io/flutter/run/bazelTest/BazelTestFields.java

```diff
@@ -0,0 +1,252 @@
+    if (StringUtil.isEmptyOrSpaces(bazelTarget)) {
+      throw new RuntimeConfigurationError(FlutterBundle.message("flutter.run.bazel.noTargetSet"));
+    }
+    // check that the bazel target starts with "//"
+    else if (!bazelTarget.startsWith("//")) {
+      throw new RuntimeConfigurationError(FlutterBundle.message("flutter.run.bazel.startWithSlashSlash"));
+    }
+  }
+
+  /**
+   * Returns the command to use to launch the Flutter app. (Via running the Bazel target.)
+   */
+  GeneralCommandLine getLaunchCommand(@NotNull Project project,
+                                      @Nullable FlutterDevice device,
+                                      @NotNull RunMode mode)
+    throws ExecutionException {
+    try {
+      checkRunnable(project);
+    }
+    catch (RuntimeConfigurationError e) {
+      throw new ExecutionException(e);
+    }
+
+    final VirtualFile appDir = MainFile.verify(entryFile, project).get().getAppDir();
+
+    final String launchingScript = getLaunchingScript();
+    assert launchingScript != null; // already checked
+
+    final String target = getBazelTarget();
+    assert target != null; // already checked
+
+    //final String additionalArgs = getAdditionalArgs();
+
+    final GeneralCommandLine commandLine = new GeneralCommandLine().withWorkDirectory(appDir.getPath());
+    commandLine.setCharset(CharsetToolkit.UTF8_CHARSET);
+    commandLine.setExePath(FileUtil.toSystemDependentName(launchingScript));
+
+    // Set the mode.
+    if (mode != RunMode.DEBUG) {
+      commandLine.addParameters("--define", "flutter_build_mode=" + mode.name());
+    }
+
+
+    //// User specified additional arguments.
+    //final CommandLineTokenizer argumentsTokenizer = new CommandLineTokenizer(StringUtil.notNullize(additionalArgs));
+    //while (argumentsTokenizer.hasMoreTokens()) {
+    //  final String token = argumentsTokenizer.nextToken();
+    //  if (token.equals("--")) {
+    //    break;
+    //  }
+    //  commandLine.addParameter(token);
+    //}
+
+    commandLine.addParameter(target);
+
+    //// Pass additional args to bazel (we currently don't pass --device-id with bazel targets).
+    //commandLine.addParameter("--");
+    //
+    //// Tell the flutter command-line tools that we want a machine interface on stdio.
+    //commandLine.addParameter("--machine");
+    //
+    //if (FlutterSettings.getInstance().getPreviewDart2()) {
+    //  commandLine.addParameter("--preview-dart-2");
+    //}
+    //
+    //// Pause the app at startup in order to set breakpoints.
+    //if (mode == RunMode.DEBUG) {
+    //  commandLine.addParameter("--start-paused");
+    //}
+    //
+    //// More user-specified args.
+    //while (argumentsTokenizer.hasMoreTokens()) {
+    //  commandLine.addParameter(argumentsTokenizer.nextToken());
+    //}
+    //
+    //// Send in the deviceId.
+    //if (device != null) {
+    //  commandLine.addParameter("-d");
+    //  commandLine.addParameter(device.deviceId());
+    //}
```

Comment:
Similarly here with all of this.  Maybe add a TODO or something if it needs to stay?